clusterloader2/pkg/measurement/common/slos/slo_measurement.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package slos
+
+import (
+	"fmt"
+	"k8s.io/klog"
+	"k8s.io/perf-tests/clusterloader2/pkg/errors"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+)
+
+const (
+	sloMeasurementName = "SloMeasurement"
+)
+
+func init() {
+
+	var sloMeasurement sloMeasurement
+	var err error
+	
+	if sloMeasurement.networkProgrammingLatency, err = measurement.CreateMeasurement("NetworkProgrammingLatency"); err != nil {
+		klog.Errorf("cannot create instance of %s: %v", "NetworkProgrammingLatency", err)
+	}
+	if sloMeasurement.dnsLookupLatency, err = measurement.CreateMeasurement("DnsLookupLatency"); err != nil {
+		klog.Errorf("cannot create instance of %s: %v", "DnsLookupLatency", err)
+	}
+
+	createSloMeasurement := func() measurement.Measurement {
+		return createSloMeasurement(sloMeasurement.networkProgrammingLatency, sloMeasurement.dnsLookupLatency)
+	}
+	if err := measurement.Register(sloMeasurementName, createSloMeasurement); err != nil {
+		klog.Fatalf("Cannot register %s: %v", sloMeasurementName, err)
+	}
+}
+
+func createSloMeasurement(networkProgrammingLatency measurement.Measurement, dnsLookupLatency measurement.Measurement) measurement.Measurement {
+	return &sloMeasurement{
+		networkProgrammingLatency: networkProgrammingLatency,
+		dnsLookupLatency:          dnsLookupLatency,
+	}
+}
+
+type sloMeasurement struct {
+	dnsLookupLatency                 measurement.Measurement
+	networkProgrammingLatency        measurement.Measurement
+	summaries                        []measurement.Summary
+	dnsLookupLatencyMetrics          *sloMeasurementOutput
+	networkProgrammingLatencyMetrics *sloMeasurementOutput
+}
+
+type sloMeasurementOutput struct {
```

Comment:
We don't need (and want) common summary - we still want to produce summaries exactly the same way we were produting them before.